src/ansys/tools/visualization_interface/backends/pyvista/pyvista.py

```diff
@@ -136,6 +158,10 @@ def __init__(
 
         self._enable_widgets = self._pl._enable_widgets
 
+        self._hover_picker = vtkPointPicker()
+        self._hover_widget = vtkHoverWidget()
```

Comment:
```py
self._hover_picker = vtkPointPicker() if self. _allow_picking else None
self._hover_widget = vtkHoverWidget() if self. _allow_hovering else None
```
Watch out with having 2 programatically unrelated variables (``self. _allow_picking`` and ``self. _allow_hovering``) to switch behaviours. You might want to define something like:

```

class PickingMode(Enum):
    """Enumerate with the available picking modes."""
    PICK = "pick"
    HOVER = "hover"
    NONE = "None"

class PyVistaBackendInterface(BaseBackend):
    ...
        
    def __init__(self, ...):
        ....
        self.__mode = PickingMode.NONE

    @property
    def _allow_picking(self):
        return self.__mode == PickingMode.PICK
    
    @_allow_picking.setter
    def _allow_picking(self, value: bool):
        if value:
            self.__mode = PickingMode.PICK
        else:
            self.__mode = PickingMode.NONE
    
    @property
    def _allow_hovering(self):
        return self.__mode == PickingMode.HOVER
    
    @_allow_hovering.setter
    def _allow_hovering(self, value: bool):
        if value:
            self.__mode = PickingMode.HOVER
        else:
            self.__mode = PickingMode.NONE
```

Reply:
This might be safer - I agree

Reply:
> ```python
> self._hover_picker = vtkPointPicker() if self. _allow_picking else None
> self._hover_widget = vtkHoverWidget() if self. _allow_hovering else None
> ```

These 2 variables are meant for hovering only, in any case i can do this:

```python
self._hover_picker = vtkPointPicker() if self. _allow_hovering else None
self._hover_widget = vtkHoverWidget() if self. _allow_hovering else None
```

Reply:
I'm not sure if I'm understanding your point regarding the behaviour switching. I missed to remove the `PickingMode` enum, is not being used anymore, this might have caused some confusion. 

Right now, the flags are only being used to either select hovering or picking:
```python
  if self._allow_picking:
      self.enable_picking()
  elif self._allow_hovering:
      self.enable_hover()
```
How are the variables switching behaviour?